File: src/main/java/model/ExtCourse.java
package model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Lob;

@Entity
public class ExtCourse extends Course {
    @Lob
    @Column(length = 1000)
    private String link;
    private String source;

    public ExtCourse(String name, String description, double price, User publisher, String link, String source) {
        super(name,description,price,publisher);
        this.link = link;
        this.source = source;
    }

    public ExtCourse() {
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getSource() {
        return source;
    }
}

File: client/src/app/course/ExtCourse.ts
import {User} from "../User";

export class ExtCourse{
  id: number;
  name: string;
  description: string;
  price: number;
  rating: number;
  publisher: User;
  link: string;
  source: string;

  constructor(id: number, name: string, description: string, price: number, rating: number, publisher: User, link: string, source: string) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
    this.rating = rating;
    this.publisher = publisher;
    this.link = link;
    this.source = source;
  }
}
